chart.js.d.ts

```diff
@@ -0,0 +1,605 @@
+      footer?:        (xLabel: string | Array<string>, yLabel: string | Array<string>, index: number, data: Object) => string | Array<string>
+      afterFooter?:   (xLabel: string | Array<string>, yLabel: string | Array<string>, index: number, data: Object) => string | Array<string>
+      // @formatter:on
+    }
+  }
+
+  // What fields does a tooltip object passed to custom contain?
+  interface Tooltip {
+
+  }
+
+  interface ChartAnimationOptions {
+    duration?: number
+    easing?: string
+    onProgress?: (args: {chart: ChartInstance, animationDetails: AnimationDetails}) => void // Check required, what fields should animationDetails have?
+    onComplete?: (args: {chart: ChartInstance, animationDetails: AnimationDetails}) => void // Check required
+  }
+
+  // Check required. Fields?
+  interface AnimationDetails {
+
+  }
+
+  interface ChartElementsOptions {
+    arc?: {
+      backgroundColor?: string
+      borderColor?: string
+      borderWidth?: number
+    }
+    line?: {
+      tension?: number
+      backgroundColor?: string
+      borderWidth?: number
+      borderColor?: string
+      borderCapStyle?: string
+      borderDash?: Array<number>
+      borderDashOffset?: number
+      borderJoinStyle?: string
+      fill?: boolean
+    }
+    fill?: {
+      radius?: number
+      pointStyle?: string
+      backgroundColor?: string
+      borderWidth?: number
+      borderColor?: string
+      hitRadius?: number
+      hoverRadius?: number
+      hoverBorderWidth?: number
+    }
+    rectangle?: {
+      backgroundColor?: string
+      borderWidth?: number
+      borderColor?: string
+      borderSkipped?: string
+    }
+  }
+
+  interface Element {
+
+  }
+
+  class ChartInstance {
+    destroy(): void;
+    update(duration?: number, lazy?: boolean): void;
+    render(duration?: number, lazy?: boolean): void;
+    stop(): void;
+    resize(): void;
+    clear(): void;
+    toBase64Image(): string;
+    generateLegend(): string;
+    getElementAtEvent(e: Event): Element; // Check required. What does element return type contain? is Event correct here?
+    getElementsAtEvent(e: Event): Array<Element>; // Check required
+    getDatasetAtEvent(e: Event): Array<Element>; // Check required
+  }
+
+  interface LineChartData extends ChartData {
+    datasets?: Array<ChartDataSet & {
+      label?: string
+      fill?: boolean
+      backgroundColor?: string
+      borderColor?: string | Array<string>
+      borderCapStyle?: string
+      borderDash?: Array<number>
+      borderDashOffset?: number
+      borderJoinStyle?: string
+      pointBorderColor?: string | Array<string>
+      pointBackgroundColor?: string | Array<string>
+      pointBorderWidth?: number | Array<number>
+      pointHoverRadius?: number | Array<number>
+      pointHoverBackgroundColor?: string | Array<string>
+      pointHoverBorderColor?: string | Array<string>
+      pointHoverBorderWidth?: number | Array<number>
+      tension?: number
+      yAxisID?: string
```

Comment:
xAxisID seems to be missing in the docs, should I just add that fix to this pull request as well?